test/orange/detail/UnitIndexer.test.cc

```diff
@@ -17,11 +19,38 @@ namespace celeritas
 {
 namespace test
 {
+class UnitIndexerTest : public Test
+{
+  public:
+    using CollectionHostRef
+        = UnitIndexerData<Ownership::const_reference, MemSpace::host>;
+    using VecSize = std::vector<size_type>;
+
+    const CollectionHostRef& host_ref() const { return data_.host(); }
+
+    void set_data(VecSize surfaces, VecSize volumes)
+    {
+        UnitIndexerData<Ownership::value, MemSpace::host> data;
+
+        auto cb_s = make_builder(&data.surfaces);
+        cb_s.insert_back(surfaces.begin(), surfaces.end());
+
+        auto cb_v = make_builder(&data.volumes);
+        cb_v.insert_back(volumes.begin(), volumes.end());
+
+        data_ = CollectionMirror<UnitIndexerData>{std::move(data)};
+    }
+
+  protected:
+    CollectionMirror<UnitIndexerData> data_;
+};
+
 //---------------------------------------------------------------------------//
 
-TEST(UnitIndexerTest, single)
+TEST_F(UnitIndexerTest, single)
 {
-    UnitIndexer indexer({4}, {10});
+    set_data({0, 4}, {0, 10});
+    UnitIndexer indexer(host_ref());
```

Comment:
Explicitly calling `this->` for member functions improves readability (even in unit tests):
```suggestion
    this->set_data({0, 4}, {0, 10});
    UnitIndexer indexer(this->host_ref());
```